akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -669,19 +677,52 @@ private[stream] final class VirtualProcessor[T] extends AtomicReference[AnyRef]
 
   private def noDemand = "spec violation: onNext was signaled from upstream without demand"
 
-  private class WrappedSubscription(real: Subscription) extends Subscription {
+  object WrappedSubscription {
+    sealed trait SubscriptionState { def demand: Long }
+    case object PassThrough extends SubscriptionState { override def demand: Long = 0 }
+    case class Buffering(demand: Long) extends SubscriptionState
+  }
+
+  // Extdending AtomicReference to make the hot memory location share the same cache line with the Subscription
+  private class WrappedSubscription(real: Subscription)
+    extends AtomicReference[WrappedSubscription.SubscriptionState] with Subscription {
+    import WrappedSubscription._
+
+    set(Buffering(0))
```

Comment:
would it be better to `extends AtomicReference[WrappedSubscription.SubscriptionState](Buffering(0))`?